lib/perl/Genome/Model/Tools/Gatk/HaplotypeCaller.pm

```diff
@@ -95,4 +95,19 @@ sub _shellcmd_extra_params {
     );
 }
 
+sub _cmdline_args {
+    my $self = shift;
+
+    my @args = $self->SUPER::_cmdline_args(@_);
+
+    if ($self->version eq '3.4' and
+            $self->output_vcf =~ /g\.vcf\.gz$/ and
+            $self->emit_reference_confidence eq 'GVCF') {
+        #Handle a bug in GATK 3.4 by including these deprecated parameters.
+        push @args, qw(-variant_index_type LINEAR -variant_index_parameter 128000);
```

Comment:
They don't need to vary.  Here's the error message if they're omitted; it specifies these exact parameters and values:
```
##### ERROR ------------------------------------------------------------------------------------------
##### ERROR A USER ERROR has occurred (version debian/3.4-2-0-g9765327): 
##### ERROR
##### ERROR This means that one or more arguments or inputs in your command are incorrect.
##### ERROR The error message below tells you what is the problem.
##### ERROR
##### ERROR If the problem is an invalid argument, please check the online documentation guide
##### ERROR (or rerun your command with --help) to view allowable command-line arguments for this tool.
##### ERROR
##### ERROR Visit our website and forum for extensive documentation and answers to 
##### ERROR commonly asked questions http://www.broadinstitute.org/gatk
##### ERROR
##### ERROR Please do NOT post this error to the GATK forum unless you have really tried to fix it yourself.
##### ERROR
##### ERROR MESSAGE: GVCF output requires a specific indexing strategy.  Please re-run including the arguments -variant_index_type LINEAR -variant_index_parameter 128000.
##### ERROR ------------------------------------------------------------------------------------------
```